### PR TITLE
Add error() for displaying errors from client code

### PR DESCRIPTION
**Pull request by shadowspawn**
# Pull Request

## Problem

People adding extra error checks do not have a way to take advantage of the error configuration built into Commander such as: `.configureOutput()`, `.exitOverride()`, `.showHelpAfterError()`.

There have not been many requests for this, but we do now have quite rich error handling which clients can use for their own errors too.

Related: #1358 #1632

## Solution

Provide a low level routine that is also used by Commander itself. Refactor private `._displayError()` into public `.error()`. Change the signature to make the error id `code` and `exitCode` optional so can be called with just a message.

```
if (cmd.opts().good && cmd.opts().bad)
  cmd.error('error: specify just one of "--good" and "--bad"');
```

## ChangeLog

- add `error()` for generating errors from client code just like Commander generated errors, with support for  `.configureOutput()`, `.exitOverride()`, and `.showHelpAfterError()`

### CODE REVIEW

**Review by abetomo — APPROVED** [review]
awesome!

**Comment by shadowspawn** [comment]
Thanks @abetomo 
I will do another prerelease soon. No other changes in mind!

**Comment by shadowspawn** [comment]
`.error()` is included in prerelease Commander v9.0.0-1.

File: Readme.md [hunk 1]
```diff
@@ -1003,6 +1004,18 @@ the inspector port is incremented by 1 for the spawned subcommand.
...
 
+### Display error
+
+This routine is available to invoke the Commander error handling for your own error conditions. (See also the next section about exit handling.)
+
+As well as the error message, you can optionally specify the `exitCode` (used with `process.exit`)
+and `code` (used with `CommanderError`).
+
+```js
+program.exit('Password must be longer than four characters');
```
Comment (hildjj) [inline Readme.md:1015]: Did you mean `error` here instead of `exit`?

Comment (shadowspawn) [reply Readme.md:1015]: Oh drat! Did the README last, and missed that typo. Thanks.

Comment (shadowspawn) [reply Readme.md:1015]: Fixed in  #1676

**Comment by shadowspawn** [comment]
Commander v9 has been released.

**Pull request merged**